File: axemas-android/app/src/main/java/it/axant/application/HomeSectionController.java
package it.axant.application;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;

import org.json.JSONException;
import org.json.JSONObject;

import it.axant.axemas.AXMActivity;
import it.axant.axemas.AXMSectionController;
import it.axant.axemas.JavascriptBridge;
import it.axant.axemas.NavigationSectionsManager;
import it.axant.axemas.SectionFragment;


public class HomeSectionController extends AXMSectionController {

    public HomeSectionController(SectionFragment section) {
        super(section);
    }

    @Override
    public void sectionWillLoad() {

    this.section.getJSBridge().registerHandler("openMap", new JavascriptBridge.Handler() {
        @Override
        public void call(Object data, JavascriptBridge.Callback callback) {
            String uri = "https://maps.google.com/maps";
            Intent i = new Intent(Intent.ACTION_VIEW, Uri.parse(uri));
            section.startActivity(i);
        }
    });

        this.section.getJSBridge().registerHandler("openNativeController", new JavascriptBridge.Handler() {
            @Override
            public void call(Object data, JavascriptBridge.Callback callback) {
                JSONObject datum = new JSONObject();
                try {
                    datum.put("url", "www/index.html");
                } catch (JSONException e) {
                    e.printStackTrace();
                }
                NavigationSectionsManager.goTo(section.getActivity(), datum);
            }
        });
    }
}

File: axemas-ios/axemas/HomeSectionController.m
//
//  HomeSectionController.m
//  axemas
//
//  Created by Alessandro Molina on 4/10/14.
//  Copyright (c) 2014 AXANT. All rights reserved.
//

#import "HomeSectionController.h"
#import "WebViewJavascriptBridge.h"
#import "NavigationSectionsManager.h"
#import "MapViewController.h"

@implementation HomeSectionController

- (void)sectionWillLoad {
    [self.section.bridge registerHandler:@"openMap" handler:^(id data, WVJBResponseCallback responseCallback) {
        [NavigationSectionsManager pushController:[[MapViewController alloc] init] animated:YES];
        
        if (responseCallback) {
            responseCallback(nil);
        }
    }];
    
    [self.section.bridge registerHandler:@"openNativeController" handler:^(id data, WVJBResponseCallback responseCallback) {
        
        NSDictionary * datum  = @{@"url":@"www/index.html",
                                  @"title":@"Home"};
        [NavigationSectionsManager goto: datum animated:YES];
        
        if (responseCallback) {
            responseCallback(nil);
        }
    }];
}

@end
